skimage/measure/_find_contours_cy.pyx

```diff
@@ -93,13 +103,65 @@ def iterate_and_store(double[:, :] array,
             coords[0] += 1
             coords[1] = 0
 
-
         square_case = 0
         if (ul > level): square_case += 1
         if (ur > level): square_case += 2
         if (ll > level): square_case += 4
         if (lr > level): square_case += 8
 
+        # We need to handle missing data.
+        # (This could either be in the form of NaNs present in the input array,
+        # or could be the result of masking.)
+        # Start by counting the number of missing data values.
+        nan_count = 0
+        if npy_isnan(ul): nan_count += 1
+        if npy_isnan(ur): nan_count += 1
+        if npy_isnan(ll): nan_count += 1
+        if npy_isnan(lr): nan_count += 1
+
+        if nan_count > 1:
+            # If a square has 2 or more missing values, we cannot correctly
+            # infer the presence of any contour line segments within it;
+            # so just move to the next square.
+            continue
+        elif nan_count == 1:
+            # There is (up to symmetry) one square arrangement containing a
+            # missing value in which we can unambiguously draw an isoline
+            # segment:
+            # The arrangement +-
+            #                 -x (where x denotes a missing value)
```

Comment:
Is this true though? The way I understand this, the missing value above might either be + or - corresponding to the cases 1 and 9. So we actually don't know what the correct iso line might be.

Also could you elaborate what you mean with "after symmetry"?